contracts/exchange-forwarder/contracts/src/MixinExchangeWrapper.sol

```diff
@@ -223,24 +223,96 @@ contract MixinExchangeWrapper {
             // Account for the ERC20Bridge transfering more of the maker asset than expected.
             if (makerAssetProxyId == erc20BridgeProxyId) {
                 uint256 balanceAfter = IERC20Token(tokenAddress).balanceOf(address(this));
-                makerAssetAcquiredAmount = LibSafeMath.max256(
+                sellFillResults.makerAssetAcquiredAmount = LibSafeMath.max256(
                     balanceAfter.safeSub(balanceBefore),
-                    makerAssetAcquiredAmount
+                    sellFillResults.makerAssetAcquiredAmount
                 );
             }
 
-            orders[i].makerAssetData.transferOut(makerAssetAcquiredAmount);
+            orders[i].makerAssetData.transferOut(sellFillResults.makerAssetAcquiredAmount);
 
             totalWethSpentAmount = totalWethSpentAmount
-                .safeAdd(wethSpentAmount);
+                .safeAdd(sellFillResults.wethSpentAmount)
+                .safeAdd(sellFillResults.protocolFeePaid);
             totalMakerAssetAcquiredAmount = totalMakerAssetAcquiredAmount
-                .safeAdd(makerAssetAcquiredAmount);
+                .safeAdd(sellFillResults.makerAssetAcquiredAmount);
+
+            // Stop execution if the entire amount of WETH has been sold
+            if (totalWethSpentAmount >= wethSellAmount) {
+                break;
+            }
+        }
+    }
+
+    /// @dev Synchronously executes multiple calls of fillOrder until total amount of WETH (exclusive of protocol fee)
+    ///      has been sold by taker.
+    /// @param orders Array of order specifications.
+    /// @param wethSellAmount Desired amount of WETH to sell.
+    /// @param signatures Proofs that orders have been signed by makers.
+    /// @return totalWethSpentAmount Total amount of WETH spent on the given orders.
+    /// @return totalMakerAssetAcquiredAmount Total amount of maker asset acquired from the given orders.
+    function _marketSellExactAmountNoThrow(
+        LibOrder.Order[] memory orders,
+        uint256 wethSellAmount,
+        bytes[] memory signatures
+    )
+        internal
+        returns (
+            uint256 totalWethSpentAmount,
+            uint256 totalMakerAssetAcquiredAmount
+        )
+    {
+        bytes4 erc20BridgeProxyId = IAssetData(address(0)).ERC20Bridge.selector;
```

Comment:
I don't think there's any benefit to putting this in a variable. idk why we do it in the other function either.

Reply:
i kinda like it for readability, but we can put it in a constant